new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/WLSTUtils.java

```diff
@@ -0,0 +1,207 @@
+
+    logger.info("Running a Kubernetes job to execute WLST script");
+    assertDoesNotThrow(()
+        -> createWLSTJob(wlstJobName, wlstScriptConfigMapName, namespace, jobCreationContainer),
+        "WLST execution failed");
+  }
+
+  /**
+   * Create a job to execute WLST script.
+   *
+   * @param uniqueName a unique job name
+   * @param wlstScriptConfigMapName configmap holding wlst script file
+   * @param namespace name of the namespace in which the job is created
+   * @param jobContainer V1Container with job commands to execute WLST script
+   * @throws ApiException when Kubernetes cluster query fails
+   */
+  public static void createWLSTJob(String uniqueName, String wlstScriptConfigMapName, String namespace,
+      V1Container jobContainer) throws ApiException {
+    logger.info("Running Kubernetes job to execute WLST script");
+
+    V1Job jobBody = new V1Job()
+        .metadata(
+            new V1ObjectMeta()
+                .name(uniqueName)
+                .namespace(namespace))
+        .spec(new V1JobSpec()
+            .backoffLimit(0) // try only once
+            .template(new V1PodTemplateSpec()
+                .spec(new V1PodSpec()
+                    .restartPolicy("Never")
+                    .containers(Arrays.asList(jobContainer
+                        .name("execute-wlst-container")
+                        .image(image)
+                        .imagePullPolicy("IfNotPresent")
+                        .volumeMounts(Arrays.asList(
+                            new V1VolumeMount()
+                                .name("wlst-job-cm-volume") // wlst script volume
+                                .mountPath(MOUNT_POINT))))) // mounted under /sctipts inside pod
+                    .volumes(Arrays.asList(new V1Volume()
+                        .name("wlst-job-cm-volume") // deployment scripts volume
+                        .configMap(new V1ConfigMapVolumeSource()
+                            .name(wlstScriptConfigMapName)))) //config map containing wlst script
+                    .imagePullSecrets(isUseSecret ? Arrays.asList(
+                        new V1LocalObjectReference()
+                            .name(OCR_SECRET_NAME))
+                        : null))));
+    String jobName = assertDoesNotThrow(()
+        -> createNamespacedJob(jobBody), "Failed to create WLST execution Job");
+
+    logger.info("Checking if the WLST job {0} completed in namespace {1}",
+        jobName, namespace);
+    withStandardRetryPolicy
+        .conditionEvaluationListener(
+            condition -> logger.info("Waiting for job {0} to be completed in namespace {1} "
+                + "(elapsed time {2} ms, remaining time {3} ms)",
+                jobName,
+                namespace,
+                condition.getElapsedTimeInMS(),
+                condition.getRemainingTimeInMS()))
+        .until(jobCompleted(jobName, null, namespace));
+
+    // check job status and fail test if the job failed to execute WLST
```

Comment:
Will the Job fail if the corresponding WLST script fails ?

Reply:
Yes if there are any exceptions or cannot connect to the admin server the job will fail.
Of course whoever calling this utility by supplying py script will have to make sure that they exit the WLST script with exit code 1 when something goes wrong.